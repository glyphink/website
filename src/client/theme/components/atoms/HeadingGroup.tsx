/**
 * @file Atoms - HeadingGroup
 * @module website/components/atoms/HeadingGroup
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type ForwardRefExoticComponent,
  type JSX
} from 'react'
import type HeadingGroupProps from './HeadingGroup.props.mts'

/**
 * A heading and related content.
 *
 * @see https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/hgroup
 * @see {@linkcode ForwardRefExoticComponent}
 * @see {@linkcode HeadingGroupProps}
 *
 * @category
 *  atoms
 *
 * @const {ForwardRefExoticComponent<HeadingGroupProps>} HeadingGroup
 */
const HeadingGroup: ForwardRefExoticComponent<HeadingGroupProps> = forwardRef<
  HTMLElement,
  HeadingGroupProps
>(
  /**
   * @param {Props<HeadingGroupProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<hgroup>` element
   */
  (
    props: Props<HeadingGroupProps>,
    ref: ForwardedRef<HTMLElement>
  ): JSX.Element => {
    return <hgroup {...props} ref={ref} />
  }
)

HeadingGroup.displayName = 'HeadingGroup'

export default HeadingGroup
