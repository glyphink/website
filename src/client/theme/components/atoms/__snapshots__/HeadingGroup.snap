// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/HeadingGroup > should render <hgroup> 1`] = `
<hgroup>
  <h1>
    Hello, world 
    <span
      aria-label="world emoji"
      role="img"
    >
      ðŸŒŽ
    </span>
    <span
      aria-label="wave emoji"
      role="img"
    >
      ðŸ‘‹
    </span>
  </h1>
  <p>
    The quick brown fox jumped over the lazy dog
  </p>
</hgroup>
`;
